runtime/src/mm/cpp/ObjectFactory.hpp

```diff
@@ -245,35 +254,104 @@ class ObjectFactoryStorage : private Pinned {
         }
     }
 
-    KStdUniquePtr<Node> root_;
+    unique_ptr<Node> root_;
     Node* last_ = nullptr;
     SpinLock mutex_;
 };
 
+class SimpleAllocator {
+public:
+    void* Alloc(size_t size, size_t alignment) noexcept { return konanAllocAlignedMemory(size, alignment); }
+
+    static void Free(void* instance) noexcept { konanFreeMemory(instance); }
+};
+
+template <typename BaseAllocator, typename GC>
+class AllocatorWithGC {
+public:
+    AllocatorWithGC(BaseAllocator base, GC& gc) noexcept : base_(std::move(base)), gc_(gc) {}
+
+    void* Alloc(size_t size, size_t alignment) noexcept {
+        gc_.SafePointAllocation(size);
+        void* ptr = base_.Alloc(size, alignment);
```

Comment:
Both `AllocatorWithGC` and `BaseAllocator` use the same name for `Alloc` methods with clearly different semantics -- the former aborts if can't alloc, while the latter returns `nullptr` in this case.
It makes some sense to distinguish these methods (using different names or whatever).

Reply:
Good point, thanks! Moved panicking back into `ObjectFactoryStorage`.